compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1155,34 +1164,82 @@ TR::Register *OMR::Power::TreeEvaluator::iternaryEvaluator(TR::Node *node, TR::C
                                            cmp1Reg, cmp2Reg);
                }
             }
-
-         generateTrg1Src1Instruction(cg, move_opcode, node, resultReg, trueReg);
+         if (two_reg)
+            {
+            generateTrg1Src1Instruction(cg, move_opcode, node, resultReg->getHighOrder(), trueReg->getHighOrder());
+            generateTrg1Src1Instruction(cg, move_opcode, node, resultReg->getLowOrder(), trueReg->getLowOrder());
+            }
+         else
+            {
+            generateTrg1Src1Instruction(cg, move_opcode, node, resultReg, trueReg);
+            }
          generateConditionalBranchInstruction(cg, branch_opcode, node, doneLabel, ccr);
-         generateTrg1Src1Instruction(cg, move_opcode, node, resultReg, falseReg);
+         if (two_reg)
+            {
+            generateTrg1Src1Instruction(cg, move_opcode, node, resultReg->getHighOrder(), falseReg->getHighOrder());
+            generateTrg1Src1Instruction(cg, move_opcode, node, resultReg->getLowOrder(), falseReg->getLowOrder());
+            }
+         else
+            {
+            generateTrg1Src1Instruction(cg, move_opcode, node, resultReg, falseReg);
+            }
          }
       else if (compare_type.isFloatingPoint())
          {
          cmp2Reg = cg->evaluate(firstChild->getSecondChild());
          if (!skip_compare)
             generateTrg1Src2Instruction(cg, TR::InstOpCode::fcmpu, node, ccr, cmp1Reg, cmp2Reg);
-         generateTrg1Src1Instruction(cg, move_opcode, node, resultReg, trueReg);
+
+         if (two_reg)
```

Comment:
FloatingPoint will never use two registers (I think) so I would rather remove it. The same at line 1208.

Reply:
@gita-omr The two_reg flag applies to the 2nd/3rd child nodes where the result could be long value on 32 bit; the isFloatingPoint applies to the 1st compare child which could be a compare node of type float. e.g. 

lternaty
   dcompare
      some regload
      1.23456
   lconst -1
   lconst -2
The result of the ternary is a long (either -1 or -2 based on the double comparison) so on 32bit a reg pair is needed.

Reply:
got it